pkg/vmprovider/providers/vsphere2/network/network.go

```diff
@@ -0,0 +1,631 @@
+	if err != nil {
+		return nil, err
+	}
+
+	return netIfToResult(vimClient, netIf), nil
+}
+
+func netIfToResult(
+	vimClient *vim25.Client,
+	netIf *netopv1alpha1.NetworkInterface) *NetworkInterfaceResult {
+
+	ipConfigs := make([]NetworkInterfaceIPConfig, 0, len(netIf.Status.IPConfigs))
+	for _, ip := range netIf.Status.IPConfigs {
+		ipConfig := NetworkInterfaceIPConfig{
+			IPCIDR:  ipCIDRNotation(ip.IP, ip.SubnetMask, ip.IPFamily == netopv1alpha1.IPv4Protocol),
+			IsIPv4:  ip.IPFamily == netopv1alpha1.IPv4Protocol,
+			Gateway: ip.Gateway,
+		}
+		ipConfigs = append(ipConfigs, ipConfig)
+	}
+
+	pgObjRef := vimtypes.ManagedObjectReference{
+		Type:  "DistributedVirtualPortgroup",
+		Value: netIf.Status.NetworkID,
+	}
+
+	return &NetworkInterfaceResult{
+		IPConfigs:  ipConfigs,
+		MacAddress: netIf.Status.MacAddress, // Not set by NetOP.
+		ExternalID: netIf.Status.ExternalID, // Ditto.
+		NetworkID:  netIf.Status.NetworkID,
+		Backing:    object.NewDistributedVirtualPortgroup(vimClient, pgObjRef),
+	}
+}
+
+func findNetOPCondition(
+	netIf *netopv1alpha1.NetworkInterface,
+	condType netopv1alpha1.NetworkInterfaceConditionType) *netopv1alpha1.NetworkInterfaceCondition {
+
+	for i := range netIf.Status.Conditions {
+		if netIf.Status.Conditions[i].Type == condType {
+			return &netIf.Status.Conditions[i]
+		}
+	}
+	return nil
+}
+
+func waitForReadyNetworkInterface(
+	vmCtx context.VirtualMachineContextA2,
+	client ctrlruntime.Client,
+	name string) (*netopv1alpha1.NetworkInterface, error) {
+
+	netIf := &netopv1alpha1.NetworkInterface{}
+	netIfKey := types.NamespacedName{Namespace: vmCtx.VM.Namespace, Name: name}
+
+	// TODO: Watch() this type instead.
+	err := wait.PollImmediate(retryInterval, RetryTimeout, func() (bool, error) {
+		if err := client.Get(vmCtx, netIfKey, netIf); err != nil {
+			return false, ctrlruntime.IgnoreNotFound(err)
+		}
+
+		cond := findNetOPCondition(netIf, netopv1alpha1.NetworkInterfaceReady)
+		return cond != nil && cond.Status == corev1.ConditionTrue, nil
+	})
+
+	if err != nil {
+		if errors.Is(err, wait.ErrWaitTimeout) {
+			// Try to return a more meaningful error when timed out.
+			if cond := findNetOPCondition(netIf, netopv1alpha1.NetworkInterfaceFailure); cond != nil && cond.Status == corev1.ConditionTrue {
+				return nil, fmt.Errorf("network interface failure: %s - %s", cond.Reason, cond.Message)
+			}
+			if cond := findNetOPCondition(netIf, netopv1alpha1.NetworkInterfaceReady); cond != nil && cond.Status == corev1.ConditionFalse {
+				return nil, fmt.Errorf("network interface is not ready: %s - %s", cond.Reason, cond.Message)
+			}
+			return nil, fmt.Errorf("network interface is not ready yet")
+		}
+
+		return nil, err
+	}
+
+	return netIf, nil
+}
+
+// NCPCRName returns the name to be used for the NCP VirtualNetworkInterface CR.
+func NCPCRName(vmName, networkName, interfaceName string, isV1A1 bool) string {
+	var name string
+
+	if isV1A1 {
```

Comment:
Is this referring to the vmop v1a1/v1a2?. If so, should we pass in the FSS val here to NCPCRName(..,.., lib.Isv1a2enabled()) in line 411.

Reply:
No. This is referring to if the VM was a prior a v1a1 VM which means we'll netIF CRDs with the prior naming convention. The existing v1a1 naming convention was ad-hoc and limiting (like only allow one per network) that I want to get away from, especially now that we have a interface name in the spec which was a prototype oversight. If the VM already has netIF CRDs we need to continue to use those and not create new ones.

This is part of the larger issue to ensure existing VMs continue to be reconciled properly post the switch to v1a2.